src/ctapipe/calib/camera/extractor.py

```diff
@@ -0,0 +1,261 @@
+"""
+Extraction algorithms to compute the statistics from a chunk of images
+"""
+
+__all__ = [
+    "StatisticsExtractor",
+    "PlainExtractor",
+    "SigmaClippingExtractor",
+]
+
+from abc import abstractmethod
+
+import numpy as np
+from astropy.stats import sigma_clipped_stats
+
+from ctapipe.containers import StatisticsContainer
+from ctapipe.core import TelescopeComponent
+from ctapipe.core.traits import (
+    Int,
+    List,
+)
+
+
+class StatisticsExtractor(TelescopeComponent):
+    """
+    Base component to handle the extraction of the statistics from a dl1 table
+    containing charges, peak times and/or charge variances (images).
+    """
+
+    chunk_size = Int(
```

Comment:
Since you were talking also about other approaches to extract these values (rolling mean for example),
I think the `chunk_size` argument shouldn't be here in the base class.

The baseclass should only define the API that is common for all extractors.

Reply:
I'm not sure if we really need any rolling mean, since one could in principle set the `chunk_shift` to 1. Currently, I can't think of more extractors that should be implemented. Do you have some in mind that we should consider?